landoapi/revisions.py

```diff
@@ -118,3 +119,26 @@ def check_author_planned_changes(*, revision, **kwargs):
         return None
 
     return "The author has indicated they are planning changes to this revision."
+
+
+def revision_is_secure(revision, phabclient):
+    """Does the given revision contain security-sensitive data?
+
+    Such revisions should be handled according to the Security Bug Approval Process.
+    See https://wiki.mozilla.org/Security/Bug_Approval_Process.
+
+    Args:
+        revision: The 'data' element from a Phabricator API response containing
+            revision data.
+        phabclient: A PhabricatorClient instance.
+    """
+    secure_project_phid = get_secure_project_phid(phabclient)
```

Comment:
Every time this is called we go across the network to the redis cache at minimum, I'd much rather we leave things where the phid is grabbed once as part of the request and then passed around. Making this take the `phid` rather than a full phabclient means it's a pure function with only basic types going in and out as well.

Particularly I like to avoid sending phabclients down into functions unless there's a ton of value, to avoid the creep we had before where everything was layered super deep and calls were being made to phabricator all over the place. The general way things are laid out now is to call to external systems in the top level request most of the time and then pass just basic python types around.

The help functions on phabclient, such as `expect` are static methods, which again are pure functions, so you can make use of them directly off of the class `PhabricatorClient.expect(...)` without needing the instantiated client.

Reply:
Changed.